eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala

```diff
@@ -290,48 +303,63 @@ case class InvoiceResult(
     paymentHash: Sha256Digest,
     expiry: FiniteDuration)
 
+case class PaymentId(value: String) {
+  override def toString: String = value
+}
+
 case class PaymentRequest(
     prefix: LnHumanReadablePart,
-    amount: Option[MilliSatoshis],
     timestamp: Long,
     nodeId: NodeId,
-    tags: Vector[JsObject],
-    signature: LnInvoiceSignature)
+    serialized: String,
+    description: String,
+    paymentHash: Sha256Digest,
+    expiry: FiniteDuration,
+    amount: Option[MilliSatoshis])
 
 case class PaymentResult(
     id: String,
+    parentId: String,
     paymentHash: Sha256Digest,
-    preimage: Option[PaymentPreimage],
-    amountMsat: MilliSatoshis,
+    amount: MilliSatoshis,
+    targetNodeId: NodeId,
     createdAt: FiniteDuration,
-    completedAt: Option[FiniteDuration],
+    paymentRequest: Option[PaymentRequest],
     status: PaymentStatus)
 
 case class ReceivedPaymentResult(
-    paymentHash: Sha256Digest,
-    amountMsat: MilliSatoshis,
-    receivedAt: FiniteDuration)
+    paymentRequest: Option[PaymentRequest],
+    paymentPreimage: String,
+    createdAt: FiniteDuration,
+    status: PaymentStatus)
 
 sealed trait PaymentStatus
 
-object PaymentStatus {
-  case object PENDING extends PaymentStatus
-  case object SUCCEEDED extends PaymentStatus
-  case object FAILED extends PaymentStatus
-
-  def apply(s: String): PaymentStatus = s match {
-    case "PENDING"   => PENDING
-    case "SUCCEEDED" => SUCCEEDED
-    case "FAILED"    => FAILED
-    case err =>
-      throw new IllegalArgumentException(
-        s"Unknown payment status code `${err}`")
-  }
-}
+case object PaymentPending extends PaymentStatus
 
-case class PaymentId(value: String) {
-  override def toString: String = value
-}
+case class PaymentSent(
+    paymentPreimage: PaymentPreimage,
+    feesPaid: MilliSatoshis,
+    route: Seq[Hop],
+    completedAt: FiniteDuration)
+    extends PaymentStatus
+
+case class Hop(
+    nodeId: NodeId,
+    nextNodeId: NodeId,
+    shortChannelId: ShortChannelId)
+
+case class PaymentFailed(failures: Seq[PaymentFailure]) extends PaymentStatus
+
+case class PaymentFailureType(name: String)
```

Comment:
How many different `PaymentFailureType` is there? We should probably enumerate them with an ADT so we handle these in a more robust way.